subsys/logging/sys_log_net.c

```diff
@@ -0,0 +1,224 @@
+/*
+ * Copyright (c) 2018 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <logging/sys_log.h>
+
+#include <net/net_pkt.h>
+#include <net/net_context.h>
+
+/* Set this to 1 if you want to see what is being sent to server */
+#define DEBUG_PRINTING 0
+
+static struct net_context *ctx;
+static struct sockaddr server_addr;
+
+/* As there is no way to figure out these values in the hook
+ * function, use some pre-defined values.
+ */
+static const int facility = 16; /* local0 */
+static const int severity = 6; /* info */
+
+#define DATE_EPOCH "1970-01-01T00:00:00.000000-00:00"
+static char date[sizeof(DATE_EPOCH)];
+
+#if defined(CONFIG_NET_IPV6) || CONFIG_NET_HOSTNAME_ENABLE
+#define MAX_HOSTNAME_LEN NET_IPV6_ADDR_LEN
+#else
+#define MAX_HOSTNAME_LEN NET_IPV4_ADDR_LEN
+#endif
+
+static char hostname[MAX_HOSTNAME_LEN + 1];
+
+NET_PKT_SLAB_DEFINE(syslog_tx_pkts, CONFIG_SYS_LOG_BACKEND_NET_MAX_BUF);
+NET_BUF_POOL_DEFINE(syslog_tx_bufs, CONFIG_SYS_LOG_BACKEND_NET_MAX_BUF,
+		    CONFIG_SYS_LOG_BACKEND_NET_MAX_BUF_SIZE,
+		    CONFIG_NET_BUF_USER_DATA_SIZE, NULL);
+
+static struct k_mem_slab *get_tx_slab(void)
+{
+	return &syslog_tx_pkts;
+}
+
+struct net_buf_pool *get_data_pool(void)
+{
+	return &syslog_tx_bufs;
+}
+
+static void fill_header(struct net_buf *buf)
+{
+	snprintk(net_buf_tail(buf),
+		 net_buf_tailroom(buf),
+		 "<%d>1 %s %s - - - - ",
+		 facility * 8 + severity,
+		 date,
+		 hostname);
+
+	net_buf_add(buf, strlen(buf->data));
+}
+
+static void syslog_hook_net(const char *fmt, ...)
+{
+	struct net_pkt *pkt;
+	struct net_buf *frag;
+	u8_t *ptr;
+	va_list vargs;
+	int ret, len;
+
+	pkt = net_pkt_get_tx(ctx, K_NO_WAIT);
+	if (!pkt) {
+		return;
+	}
+
+	frag = net_pkt_get_data(ctx, K_NO_WAIT);
+	if (!frag) {
+		net_pkt_unref(pkt);
+		return;
+	}
+
+	net_pkt_frag_add(pkt, frag);
+
+	fill_header(frag);
+
+	va_start(vargs, fmt);
+
+	ptr = net_buf_tail(frag);
+
+	ret = vsnprintk(ptr, (net_buf_tailroom(frag) - 1), fmt, vargs);
+	if (ret < 0) {
+		net_pkt_unref(pkt);
+		return;
+	}
+
+	va_end(vargs);
+
+	len = strlen(ptr);
```

Comment:
Isn't this redundant? You already have the string length stored in the ret variable.

Reply:
Yes, I realized it too after sending the PR. Will fix this.